### ISSUE TITLE
Ensure new Pipeline is receiving data

### ISSUE THREAD

**Issue by juliev0** [numaproj/numaplane#447]
# Summary

This is to solve the following problems:

- eliminate the possibility that the new Pipeline is not receiving data due to too few Kafka partitions (no Kafka partitions available)
- only transfer data to the new Pipeline after it is showing its in a healthy state (phase=Running, healthy conditions for a period of time). (note [this ](https://github.com/numaproj/numaplane/issues/445)issue should be done first or in parallel with this one)

As recommended by Derek:

1. Update "promoted" pipeline such that its scale.max=half of the pods that it's currently running (or really the `floor(half(runningpods))` such that if 1 pod is running, then it would be 0. Given that users are not supposed to have a scale.max which exceeds the number of Kafka partitions and given that Kafka wants to keep things balanced, this should distribute many if not all partitions to the new pipeline/monovertex.  
2. If new Pipeline/mv fails assessment, then Kafka will redistribute the load back to the original one (may not happen immediately) since the new one will stop acking Kafka once there's enough backpressure due to the failure.

Future possible optimization: (I can write a separate issue for this)

The new pipeline could be created with source.scale.min and max=0 and only once we are confident that it's Running with healthy conditions, then we scale the source back up.

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by juliev0** [comment]
moving to future sprint

**Comment by afugazzotto** [comment]
Set bot scale min and max to 0.
